### PR TITLE
[SYCL] Test the case when context has multiple devices

### PR DESCRIPTION
**Pull request by againull**
Test for https://github.com/intel/llvm/pull/3721

### CODE REVIEW

File: SYCL/Regression/context_with_multiple_devices.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+using namespace cl::sycl;
+
+int main() {
+  auto Plt = platform(gpu_selector{});
+  auto Devs = Plt.get_devices();
+  auto Ctx = context(Devs);
+
+  // Test the patform with 2 or more devices
+  if (Devs.size() < 2) {
```
Comment (vladimirlaz) [inline SYCL/Regression/context_with_multiple_devices.cpp]: In most configuration only one GPU device is available. Moreover when we have 2 GPU devices they can be in different platforms. So the test will never get expected configuration..
Will the test work if we create sub-devices to make sure that we have proper configuration?

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: SYCL/Regression/context_with_multiple_devices.cpp (hunk 1 above)
Comment (againull) [reply SYCL/Regression/context_with_multiple_devices.cpp]: I intentionally wanted to test the case when one context has several devices. If we have 2 devices in 2 different platforms then each device is going to be in different context and this case is different and not related to the patch intel/llvm#3721. Unfortunately, the case with sub-devices is also different.

Comment (vladimirlaz) [reply SYCL/Regression/context_with_multiple_devices.cpp]: Could you try to use NEOReadDebugKeys=1 CreateMultipleRootDevices=2?
```
$ NEOReadDebugKeys=1 CreateMultipleRootDevices=2 sycl-ls
[opencl:0] GPU : Intel(R) OpenCL HD Graphics 3.0 [21.19.19792]
[opencl:1] GPU : Intel(R) OpenCL HD Graphics 3.0 [21.19.19792]
[opencl:0] CPU : Intel(R) OpenCL 2.1 [2021.11.3.0.09_160000]
[opencl:0] ACC : Intel(R) FPGA Emulation Platform for OpenCL(TM) 1.2 [2021.11.3.0.09_160000]
[level_zero:0] GPU : Intel(R) Level-Zero 1.1 [1.1.19792]
[level_zero:1] GPU : Intel(R) Level-Zero 1.1 [1.1.19792]
[host:0] HOST: SYCL host platform 1.2 [1.2]
```

Comment (vladimirlaz) [reply SYCL/Regression/context_with_multiple_devices.cpp]: @againull could you follow up with the PR?

Comment (againull) [reply SYCL/Regression/context_with_multiple_devices.cpp]: Fixed.

File: SYCL/Regression/context_with_multiple_devices.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env NEOReadDebugKeys=1 CreateMultipleRootDevices=2 %GPU_RUN_PLACEHOLDER %t.out
+// REQUIRES: level_zero && gpu
+
```
Comment (vladimirlaz) [inline SYCL/Regression/context_with_multiple_devices.cpp]:
```suggestion
// UNSUPPORTED: gpu-intel-dg1
```
Debug options above are not supported on DG1 card

**Comment by vladimirlaz** [comment]
@againull do you have any updates with the PR?

**Comment by vladimirlaz** [comment]
@againull, could you please provide an update on the PR?

**Comment by vladimirlaz** [comment]
@againull, please reopen the PR if it is still valid

**Pull request closed without merging**